hack/generators/generate.py

```diff
@@ -0,0 +1,320 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import re
+import jinja2
+import optparse
+import logging
+import json
+
+
+def setOptions():
+
+    parser = optparse.OptionParser("%prog ")
```

Comment:
I'd like to see what @ashcrow has to say about this as well, but I know two things:

1. The `optparse` module [is deprecated since 2.7](https://docs.python.org/2/library/optparse.html) in favor of `argparse`
1. We don't use `optparse` anywhere else in our code base, we are using `argparse` (in 7 different locations)

Perhaps switch over to [`argparse`](https://docs.python.org/2/library/argparse.html#module-argparse) instead?